### PR TITLE
[editor] Fix crash in created feature editing.

### PR DESCRIPTION
**Pull request by mgsergio**
@milchakov 
@mpimenov

### CODE REVIEW

File: indexer/osm_editor.hpp [hunk 1]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
 
   enum class FeatureStatus
   {
-    Untouched,
-    Deleted,
-    Obsolete,  // The feature is obsolete when is marked for deletion via note.
-    Modified,
-    Created
+    Untouched,  // The feature is not saved in editor.
+    Deleted,    // The feture was marked as deleted.
```
Comment (mpimenov) [inline indexer/osm_editor.hpp]: s/feture/feature/

**Comment by mgsergio** [comment]
DUDE JTTP

File: indexer/osm_editor.hpp [hunk 2]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
...
+    Obsolete,   // The feature is obsolete when is marked for deletion via note.
```
Comment (mpimenov) [inline indexer/osm_editor.hpp]: when it is marked?
when it was marked?

File: indexer/osm_editor.hpp [hunk 3]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
...
+    Modified,   // The featrue is saved in editor end is differ from the original one.
```
Comment (mpimenov) [inline indexer/osm_editor.hpp]: s/featrue/feature/

Comment (mpimenov) [inline indexer/osm_editor.hpp]: s/end/and/
s/differe/different/

File: indexer/osm_editor.hpp [hunk 4]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
...
+    Created     // The feature was created by a user and is saved in editor.
+                // Note: If a feture was created by a user but not yes saved in editor
```
Comment (mpimenov) [inline indexer/osm_editor.hpp]: s/feture/feature/

Comment (mpimenov) [inline indexer/osm_editor.hpp]: Что значит "but not yes saved"?

File: indexer/osm_editor.hpp [hunk 5]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
...
+                // it's status is Untouched.
```
Comment (mpimenov) [inline indexer/osm_editor.hpp]: s/it's/its/

File: indexer/osm_editor.cpp [hunk 6]
```diff
@@ -448,10 +448,20 @@ Editor::SaveResult Editor::SaveEditedFeature(EditableMapObject const & emo)
...
   bool const wasCreatedByUser = IsCreatedFeature(fid);
-  if (wasCreatedByUser && featureStatus == FeatureStatus::Untouched)
+  if (wasCreatedByUser)
   {
     fti.m_status = FeatureStatus::Created;
     fti.m_feature.ReplaceBy(emo);
+
+    if (featureStatus == FeatureStatus::Created)
+    {
+      auto const & editedFeatureInfo = m_features[fid.m_mwmId][fid.m_index];
```
Comment (ygorshenin) [inline indexer/osm_editor.cpp]: Do you plan to check that m_features has m_mwmId and m_index?

File: indexer/osm_editor.hpp (hunk 2 above)
Comment (mpimenov) [inline indexer/osm_editor.hpp]: when marked?

File: indexer/osm_editor.hpp (hunk 4 above)
Comment (milchakov) [inline indexer/osm_editor.hpp]: s/yes/yet/

File: indexer/osm_editor.cpp (hunk 6 above)
Comment (mgsergio) [inline indexer/osm_editor.cpp]: It's guarantied by featureStatus == FeatureStatus::Created

**Comment by mgsergio** [comment]
PTAL

File: indexer/osm_editor.hpp (hunk 2 above)
Comment (mgsergio) [inline indexer/osm_editor.hpp]: Done.

File: indexer/osm_editor.hpp (hunk 3 above)
Comment (mgsergio) [inline indexer/osm_editor.hpp]: Done.

File: indexer/osm_editor.hpp (hunk 4 above)
Comment (mgsergio) [inline indexer/osm_editor.hpp]: Done.

File: indexer/osm_editor.hpp (hunk 5 above)
Comment (mgsergio) [inline indexer/osm_editor.hpp]: Done.

File: indexer/osm_editor.hpp [hunk 7]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
...
   enum class FeatureStatus
   {
-    Untouched,
-    Deleted,
-    Obsolete,  // The feature is obsolete when is marked for deletion via note.
-    Modified,
-    Created
+    Untouched,  // The feature is not saved in editor.
+    Deleted,    // The feature was marked as deleted.
+    Obsolete,   // The feature is obsolete when marked for deletion via note.
```
Comment (mpimenov) [inline indexer/osm_editor.hpp]: s/is/becomes/

**Comment by milchakov** [comment]
LGTM

File: indexer/osm_editor.hpp [hunk 8]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
...
+    Modified,   // The feature is saved in editor and differs from the original one.
+    Created     // The feature was created by a user and is saved in editor.
+                // Note: If a feture was created by a user but not yet saved in editor
```
Comment (mpimenov) [inline indexer/osm_editor.hpp]: s/feture/feature/

Comment (mpimenov) [inline indexer/osm_editor.hpp]: запятая после editor

Comment (mgsergio) [inline indexer/osm_editor.hpp]: Done.

File: indexer/osm_editor.hpp (hunk 7 above)
Comment (mpimenov) [inline indexer/osm_editor.hpp]: или вообще "The feature is marked for deleteion via note."

File: indexer/osm_editor.hpp [hunk 9]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
 
   enum class FeatureStatus
   {
-    Untouched,
-    Deleted,
-    Obsolete,  // The feature is obsolete when is marked for deletion via note.
-    Modified,
-    Created
+    Untouched,  // The feature is not saved in editor.
```
Comment (therearesomewhocallmetim) [inline indexer/osm_editor.hpp]: ... was not ...

File: indexer/osm_editor.hpp [hunk 10]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
...
+    Deleted,    // The feature was marked as deleted.
+    Obsolete,   // The feature is obsolete when marked for deletion via note.
+    Modified,   // The feature is saved in editor and differs from the original one.
```
Comment (therearesomewhocallmetim) [inline indexer/osm_editor.hpp]: was

File: indexer/osm_editor.hpp [hunk 11]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
...
+    Created     // The feature was created by a user and is saved in editor.
```
Comment (therearesomewhocallmetim) [inline indexer/osm_editor.hpp]: by the user and was

File: indexer/osm_editor.hpp (hunk 9 above)
Comment (mpimenov) [inline indexer/osm_editor.hpp]: "is" is ok

File: indexer/osm_editor.hpp [hunk 12]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
...
+                // Note: If a feature was created by a user but not yet saved in editor
```
Comment (therearesomewhocallmetim) [inline indexer/osm_editor.hpp]: the user ... but hasn't been

Comment (mgsergio) [inline indexer/osm_editor.hpp]: Done(been).

File: indexer/osm_editor.hpp [hunk 13]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
 
   enum class FeatureStatus
   {
-    Untouched,
-    Deleted,
-    Obsolete,  // The feature is obsolete when is marked for deletion via note.
-    Modified,
-    Created
+    Untouched,  // The feature is not saved in editor.
+    Deleted,    // The feature was marked as deleted.
```
Comment (mpimenov) [inline indexer/osm_editor.hpp]: s/was/is/

File: indexer/osm_editor.hpp (hunk 9 above)
Comment (therearesomewhocallmetim) [inline indexer/osm_editor.hpp]: The feature hasn't been saved in the editor.

File: indexer/osm_editor.hpp (hunk 13 above)
Comment (therearesomewhocallmetim) [inline indexer/osm_editor.hpp]: The feature has been marked as deleted.

**Comment by mgsergio** [comment]
PTAL

**Comment by therearesomewhocallmetim** [comment]
LGTM

File: indexer/osm_editor.hpp [hunk 14]
```diff
@@ -47,11 +47,13 @@ class Editor final : public MwmSet::Observer
 
   enum class FeatureStatus
   {
-    Untouched,
-    Deleted,
-    Obsolete,  // The feature is obsolete when is marked for deletion via note.
-    Modified,
-    Created
+    Untouched,  // The feature hasn't been saved in the editor.
```
Comment (mpimenov) [inline indexer/osm_editor.hpp]: ...

Comment (mpimenov) [inline indexer/osm_editor.hpp]: is читалось легче

**Comment by mpimenov** [comment]
LGTM

**Comment by ygorshenin** [comment]
LGTM, but fellows, when you'll implement synchronization for Editor?

**Comment by ygorshenin** [comment]
The code is incorrect in multithreaded environment?

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**